.changeset/deep-symbols-heal.md

```diff
@@ -0,0 +1,19 @@
+---
+"@cloudflare/vite-plugin": patch
+---
+
+Add experimental support for Workers Assets metafiles (\_headers and \_redirects) in `vite dev`.
+
+This is behind an experimental option in the cloudflare plugin configuration:
+
+```ts
+cloudflare({
+	// ...
+	experimental: { headersAndRedirectsDevModeSupport: true },
+}),
+```
+
+Currently, in this experimental mode, requests that would result in an HTML response or a 404 response will take into account the \_headers and \_redirects settings.
+Known limitation: requests for existing static assets will be served directly by Vite without considering the \_headers or \_redirects settings.
+
+A production deployment or `vite preview` already accurately supports these settings.
```

Comment:
My main worry with including it in the changelog is that it invites people to try it out. If we put it in the changelog, can we get changesets to clearly mark this as unstable? Have a look at this React Router changelog for an example of what I mean - https://github.com/remix-run/react-router/blob/main/CHANGELOG.md#unstable-changes.

Reply:
I don't think changesets can do that. I can make this look more scary and indicate it might disappear.

But I feel like it needs a changeset of some kind, right?

Reply:
I'll let you decide what's best. Just flagging up the concern.